crates/router/src/configs/settings.rs

```diff
@@ -918,3 +773,77 @@ pub struct PayPalOnboarding {
     pub partner_id: masking::Secret<String>,
     pub enabled: bool,
 }
+
+fn deserialize_hashset<'a, D, T>(deserializer: D) -> Result<HashSet<T>, D::Error>
+where
+    D: serde::Deserializer<'a>,
+    T: Eq + std::str::FromStr + std::hash::Hash,
+{
+    let value = <String>::deserialize(deserializer)?;
+    Ok(value
+        .trim()
+        .split(',')
+        .flat_map(|s| T::from_str(s.trim()))
```

Comment:
We probably don't want `flat_map` since it will cause any `from_str` errors to be silently ignored instead of being reported.